### DISCUSSION TITLE
7.3.0 cy.intercept - How to match urls which contain encoded querystring

### DISCUSSION THREAD

**Discussion by luichooy** [cypress-io/cypress#16633] [Questions and Help] [answered]
In my web app, there are two similar request in one single page:

```
// equal to  '/mclaren/ad.detail/1.0?param={"planTypeCode":7}'
https://hostname/mclaren/ad.detail/1.0?param=%7B%22planTypeCode%22:7%7D

// equal to '/mclaren/ad.detail/1.0?param={"planTypeCode":15}'
https://hostname/mclaren/ad.detail/1.0?param=%7B%22planTypeCode%22%3A15%7D
```

Glob pattern can macth all of them which is not my desire
```
cy.intercept('mclaren/ad.detail/1.0?*', req => {
    console.log(req)
})
```

How to match  them separately ?

Thanks!

**Comment by jennifer-shehane** [comment, marked as answer]
This is supported via the query matcher in cy.intercept(). Cypress parses and decodes the query string before matching, so you can target each request by its decoded param value rather than the raw encoded URL:

```js
cy.intercept({
  pathname: '/mclaren/ad.detail/1.0',
  query: { param: '{"planTypeCode":7}' },
}).as('plan7')

cy.intercept({
  pathname: '/mclaren/ad.detail/1.0',
  query: { param: '{"planTypeCode":15}' },
}).as('plan15')
```

Using pathname keeps the encoded query string out of glob path-matching. See the cy.intercept docs for the query/pathname matchers (and RegExp/function matchers for more complex cases).
